driver.go

```diff
@@ -51,26 +55,64 @@ func (d *Driver) Get(request *volume.GetRequest) (*volume.GetResponse, error) {
 
 func (d *Driver) Remove(request *volume.RemoveRequest) error {
 	log.Debugf("Request Remove: Name=%s. It will succeed regardless of the presence of the volume", request.Name)
+	overlay_path := d.overlaysCreated[request.Name]
+	err := overlay.Unmount(overlay_path)
+	if err != nil {
+		return err
+	}
+	err = os.RemoveAll(overlay_path)
+	if err != nil {
+		return err
+	}
 	delete(d.volumesCreated, request.Name)
 	return nil
 }
 
 func (d *Driver) Path(request *volume.PathRequest) (*volume.PathResponse, error) {
-	log.Debugf("Request Path: Name=%s", request.Name)
-	log.Error("Path is not implemented. Returning an error")
-	return nil, errors.New("the Path request is not implemented")
+	return &volume.PathResponse{Mountpoint: d.volumesCreated[request.Name].Mountpoint}, nil
 }
 
+/*
+TEST:
+sudo docker volume create --driver docker-on-top test101
+sudo docker run -name ubuntu_bash_101 --mount type=volume,source=test101,destination=/usr -t -i --privileged ubuntu bash
+cd ./usr
+> nex.txt #check if this file appears in overlay_path/upper directory
+sudo docker rm ubuntu_bash_101
+sudo docker volume rm test101
+# check if the /home/VolumeOverlays/test101 was deleted
+*/
 func (d *Driver) Mount(request *volume.MountRequest) (*volume.MountResponse, error) {
 	log.Debugf("Request Mount: ID=%s, Name=%s", request.ID, request.Name)
-	log.Error("Mount is not implemented. Returning an error")
-	return nil, errors.New("the Mount request is not implemented")
+	path := "/home/VolumeOverlays/" + request.Name
+	err := os.MkdirAll(path, os.ModePerm)
+	err1 := os.MkdirAll(path+"/lower", os.ModePerm)
+	err2 := os.MkdirAll(path+"/upper", os.ModePerm)
+	err3 := os.MkdirAll(path+"/work_dir", os.ModePerm)
+	if (err != nil || err1 != nil || err2 != nil || err3 != nil) && !os.IsExist(err) {
+		log.Error("ERROR in mkdir:")
+		log.Error(err)
+		return nil, err
+	}
+	_, err = overlay.Mount(path, path+"/lower", path+"/upper", 0, 0, []string{"lowerdir=" + path + "/lower,upperdir=" + path + "/upper,work_dir=" + path + "/work_dir"})
```

Comment:
Wait, there actually is a way more serious problem. What and where do you actually mount?

- `/work_dir` is something internal to overlay, which you create and hand to it and don't care about anything from that point on - that is correct and makes sense
- `/upper` is where the changes are stored, but it is different from the mountpoint! Mountpoint is _where_ you mount your overlay - where you will be able to access the virtual filesystem and modify files, while the upper dir is where your changes are stored (by the `overlay` implementation), that's not a directory you should access directly either. I see why this may still work and produce the appropriate results, but it is an unreliable behavior that depends on implementation. Don't do that - mountpoint should be a different directory.
- `/lower` is the base that you want to make a filesystem on top of. So why is it created by you as an empty directory? This is something that the user should provide (most likely, on the volume creation stage) and you should mount on top of it. Otherwise, it does not make much sense: you are making an overlay, but it is always on top of an empty directory that you yourself created.

Reply:
And, finally, it doesn't seem to work! I ran this on my machine, created a volume with the driver (which the logs confirmed) then started a container with the volume, for which the plugin was requested to mount it (which the logs confirmed) but I don't think anything was actually mounted:
1. I ran the `mount` command (without arguments) which printed the list of all current mounts but didn't find anything containing `/home/VolumeOverlays`
2. While the volume was seemingly mounted, I was able to perform `rmdir` for all lower, upper and work_dir, which, I suppose, should have resulted in an error if there was an overlay mount

I don't know whether it is an issue with the library you are using or something else but this doesn't seem to work.